### DISCUSSION TITLE
Optional nullable query parameter raise 422 error

### DISCUSSION THREAD

**Discussion by chbndrhnns** [fastapi/fastapi#6389] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import fastapi
import httpx
from pydantic import UUID4

app = fastapi.FastAPI()


@app.get("/")
async def get(my_param: UUID4 = fastapi.Query(None, nullable=True)):
    return {"my_param": my_param}


async def test_can_get():
    async with httpx.AsyncClient(app=app, base_url="http://mytesthost.com") as client:
        res = await client.get("/", params={"my_param": None})
        assert res.status_code == 200, res.content
```

### Description

* Run the test and expect it to pass
* The test fails

Error:
```
{"detail":[{"loc":["query","my_param"],"msg":"value is not a valid uuid","type":"type_error.uuid"}]}
```

### Operating System

macOS

### FastAPI Version

0.74.1

### Python Version

3.9.9

**Comment by chbndrhnns** [comment]
I found a [related](https://github.com/tiangolo/fastapi/issues/4708) issue. The reason I'd like to use a UUID type and not just `str` is to indicate what format is expected.

**Comment by ComicShrimp** [comment]
You could try to use `Optional[]` in the parameter. the final code will be something like this:
```python
import fastapi
import httpx
from pydantic import UUID4
from typing import Optional

app = fastapi.FastAPI()


@app.get("/")
async def get(my_param: Optional[UUID4] = fastapi.Query(None, nullable=True)):
    return {"my_param": my_param}


async def test_can_get():
    async with httpx.AsyncClient(app=app, base_url="http://mytesthost.com") as client:
        res = await client.get("/", params={"my_param": None})
        assert res.status_code == 200, res.content
```

That way not only the query parameter will be optional, but the variable will be optional too.
I hope that it helps !!!

**Comment by chbndrhnns** [comment]
I thought so, too. But when I run your snippet with FastAPI 0.77, it fails with a 422.

**Comment by ComicShrimp** [comment]
Well, the other thing that could cause the problem is that you're passing the parameters on the request. If the value is None, you can just omit the parameter.

```Python
import fastapi
import httpx
from pydantic import UUID4
from typing import Optional

app = fastapi.FastAPI()


@app.get("/")
async def get(my_param: Optional[UUID4] = fastapi.Query(None, nullable=True)):
    return {"my_param": my_param}


async def test_can_get():
    async with httpx.AsyncClient(app=app, base_url="http://mytesthost.com") as client:
        res = await client.get("/")
        assert res.status_code == 200, res.content
```

**Comment by chbndrhnns** [comment]
I intended to capture this `null` value and perform a special operation in the `null` case. I dropped this idea after understanding that it's also hard to differentiate between an incoming `null` and a default `None`

However, I assumed that setting `nullable` to `True` means: This field can be passed as `null`.

**Comment by chbndrhnns** [comment, marked as answer]
I understand now that the query parameters are received as a string. Sending "?some_val=null" might also cause problems if "null" is a valid option for "some_val". In this case, it's impossible to distinguish between the valid option and nothing.

**Comment by edugmes** [comment]
I have the exact same problem (except it's an UUID). Has anybody got a solution for that on Python 3.10 and Fastapi 0.88.0?
